src/ObjectGraph.es6.js

```diff
@@ -316,9 +352,17 @@ ObjectGraph.prototype.capture = function(o, opts) {
   this.q.onDone = function() {
     this.timestamp = (new Date()).getTime();
     this.initLazyData();
-    prevOnDone(this);
     opts.onDone && opts.onDone(this);
     this.busy = false;
+    this.instanceQueue.onDone = function() {
+      if (!this.q.empty()) this.q.flush();
+      else prevOnDone(this);
+    }.bind(this);
+    this.q.onDone = function() {
+      if (!this.instanceQueue.empty()) this.instanceQueue.flush();
+      else prevOnDone(this);
+    }.bind(this);
+    this.instanceQueue.flush();
   }.bind(this);
 
   // Edge case: Passed-in object is blacklisted. This is not caught by
```

Comment:
LIne 385: nit: spacing: `if ( ... )`. (I know it's not part of this CL, but let's fix it while we're here.